indico/modules/events/registration/controllers/management/__init__.py

```diff
@@ -60,7 +60,8 @@ def _checkParams(self, params):
         RHManageRegFormBase._checkParams(self, params)
         self.registration = (Registration
                              .find(Registration.id == request.view_args['registration_id'],
-                                   ~RegistrationForm.is_deleted)
+                                   ~RegistrationForm.is_deleted,
+                                   ~Registration.is_deleted)
```

Comment:
I've move this up (first the Registration conditions, then the RegistrationForm conditions)